p2panda-engine/src/operation.rs

```diff
@@ -0,0 +1,260 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+use ciborium::de::Error as CiboriumError;
+use p2panda_core::{
+    validate_backlink, validate_operation, Body, Extension, Header, Operation, OperationError,
+};
+use p2panda_store::{LogStore, OperationStore, StoreError};
+use serde::de::DeserializeOwned;
+use serde::Serialize;
+use thiserror::Error;
+
+use crate::extensions::PruneFlag;
+
+/// Encoded bytes of an operation header and optional body.
```

Comment:
It would be nice to have this in p2panda-core eventually, maybe behind a `cbor` flag